applications/connectivity_bridge/CMakeLists.txt

```diff
@@ -7,15 +7,40 @@
 cmake_minimum_required(VERSION 3.20.0)
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+find_package(Git)
 project(NONE)
 
 # NORDIC SDK APP START
 target_sources(app PRIVATE src/main.c)
 # NORDIC SDK APP END
 
+# Generate a C header file with the readme text as a string variable
+set(OUTPUT_README_HEADER_FILE "${CMAKE_BINARY_DIR}/generated/readme/readme.h")
+set(INPUT_README_FILE ${CMAKE_SOURCE_DIR}/boards/${BOARD}_readme.txt)
+if(NOT EXISTS ${INPUT_README_FILE})
+message(WARNING "${INPUT_README_FILE} does not exist")
+set(INPUT_README_FILE ${CMAKE_SOURCE_DIR}/README_TEMPLATE.txt)
+endif()
+file(READ ${INPUT_README_FILE} README_CONTENT)
+if(GIT_FOUND)
+  execute_process(
+        COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        OUTPUT_VARIABLE GIT_HASH
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  string(TIMESTAMP CURRENT_DATE "%Y-%m-%d")
+  string(APPEND README_CONTENT "\nFirmware version: ${GIT_HASH} Build date: ${CURRENT_DATE}")
+endif()
+string(REPLACE "\"" "\\\"" README_CONTENT "${README_CONTENT}")
+string(REPLACE "\n" "\\n\"\n\"" README_CONTENT "${README_CONTENT}")
+file(WRITE ${OUTPUT_README_HEADER_FILE} "static const char* file_contents = \"${README_CONTENT}\";\n\n")
```

Comment:
I would prefer if `file_contents` is not declared in `readme.h`. I think `readme.c` becomes more readable if `readme.h` only contains the string text, and `readme.c` could include it like this:

`static const char file_contents[] = { #include "readme.h" };`